api/v1beta1/azurecluster_webhook.go

```diff
@@ -128,6 +128,44 @@ func (c *AzureCluster) ValidateUpdate(oldRaw runtime.Object) error {
 		)
 	}
 
+	oldSubnetMap := make(map[string]SubnetSpec, len(old.Spec.NetworkSpec.Subnets))
+	oldSubnetIndex := make(map[string]int, len(old.Spec.NetworkSpec.Subnets))
+	for i, subnet := range old.Spec.NetworkSpec.Subnets {
+		oldSubnetMap[subnet.Name] = subnet
+		oldSubnetIndex[subnet.Name] = i
+	}
+	for i, subnet := range c.Spec.NetworkSpec.Subnets {
+		if oldSubnet, ok := oldSubnetMap[subnet.Name]; ok {
+			// Verify the CIDR blocks haven't changed for an owned Vnet.
+			// A non-owned Vnet's CIDR block can change based on what's
+			// defined in the spec vs what's been loaded from Azure directly.
+			if old.Spec.NetworkSpec.Vnet.Tags.HasOwned(old.Name) && !reflect.DeepEqual(subnet.CIDRBlocks, oldSubnet.CIDRBlocks) {
```

Comment:
how does this work when the tags haven't been set yet (they get set by the controller when we create/get the vnet the first time) we're creating a new cluster but the CIDRBlock gets defaulted in the defaulting webhook? Should this allow going from empty string to not-empty-string?

Reply:
ha just read https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/2635#discussion_r985268538 which basically answers my question

Could you please clarify this in a code comment? That's good info. And to answer your question, I agree that's acceptable too, as long as we're not blocking the webhook from defaulting.

Reply:
Sure, I'll make the comment more verbose. Thanks!

Reply:
Added